tests/statistics/stats_utils.py

```diff
@@ -0,0 +1,105 @@
+# AUTHORS file for copyright and authorship information.
+
+from collections import OrderedDict
+from datetime import datetime
+
+import pytest
+
+from pytest_pootle.factories import UserFactory
+
+from django.contrib.auth import get_user_model
+from django.db.models import Q
+
+from pootle.core.delegate import contributors
+from pootle_statistics.models import Submission, SubmissionTypes
+from pootle_statistics.utils import Contributors
+from pootle_store.models import Store
+
+
+User = get_user_model()
+
+
+def _contributors_list(contribs,
+                       projects=None, languages=None,
+                       since=None, until=None):
+    subs = []
+    for user in contribs.user_qs.distinct():
+        q = Q(submitter=user)
+        if projects:
+            q = q & Q(
+                translation_project__project__code__in=projects)
+        if languages:
+            q = q & Q(
+                translation_project__language__code__in=languages)
+        if since or until:
+            _q = Q()
+            if since:
+                _q = _q & Q(creation_time__gte=since)
+            if until:
+                _q = _q & Q(creation_time__lte=until)
+            q = q & _q
+        submissions = Submission.objects.filter(q)
+        if submissions.count():
+            subs.append((user.username, submissions.count()))
+    return OrderedDict(sorted(subs, key=lambda i: (-i[1], i[0])))
+
+
+@pytest.fixture
+def anon_submission_unit():
+    anon = User.objects.get(username="nobody")
+    unit = Store.objects.live().first().units.first()
+    old_target = unit.target
+    old_state = unit.state
+    unit.target_f = "Updated %s" % old_target
+    unit._target_updated = True
+    unit.store.record_submissions(
+        unit, old_target, old_state,
+        datetime.now(), anon,
+        SubmissionTypes.NORMAL)
+    unit.save()
+
+
+@pytest.mark.django_db
+def test_contributors_delegate():
+    contribs_class = contributors.get()
+    assert contribs_class is Contributors
+
+
+@pytest.mark.django_db
+def test_contributors_instance(member, anon_submission_unit):
+    """Contributors across the site."""
+    anon = User.objects.get(username="nobody")
+    contribs = Contributors()
+    someuser = UserFactory()
+    assert contribs.include_anon is False
+    assert contribs.language_codes is None
+    assert contribs.project_codes is None
+
+    # user_qs
+    assert list(contribs.user_qs) == list(User.objects.hide_meta())
+    assert someuser in contribs.user_qs
+    assert anon not in contribs.user_qs
+    assert (
+        sorted(contribs)
+        == sorted(contribs.contributors)
+        == sorted(
+            set(contribs.user_qs.filter(contribs.user_filters)
+                                .values_list("username", flat=True)))
+        == sorted(
+            set(contribs.user_qs.filter(submission__gt=0)
+                                .values_list("username", flat=True))))
```

Comment:
Can't this be split into separate assert calls? Also perhaps the long queries can be stored in variables, just for ease reading.